### PR TITLE
Add `return_parsed` kwarg to `s3_copy`

### PR DESCRIPTION
**Pull request by hannahilea**
Add `return_parsed` kwarg (default `true` to be non-breaking) to `s3_copy`.

Related to #293.

### CODE REVIEW

**Review by mattBrzezinski — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 - v0.11.0: The `s3_exists` and `isdir(::S3Path)` calls no longer encounter HTTP 403 (Access Denied) errors when attempting to list resources which requiring an `s3:prefix` to be specified ([#289]).
 - v0.11.1: The new keyword argument `returns` for `Base.write(fp::S3Path, ...)` determines the output returned from `write`, which can now be the raw `AWS.Response` (`returns=:response`) or the `S3Path` (`returns=:path`); this latter option returns an `S3Path` populated with the version ID of the written object (when versioning is enabled on the bucket) ([#293]).
+- v0.11.2: The new keyword argument `parse_response` for `s3_copy` enables, e.g., accessing the version ID of a copied object (when versioning is enabled on the bucket) ([#300]).
```
Comment (omus) [inline CHANGELOG.md:11]:
```suggestion
- v0.11.2: `s3_copy` supports the `parse_response` keyword allowing for access to the unparsed AWS API response ([#300]).
```

**Review by omus — APPROVED** [review]

**Comment by omus** [comment]
bors try

**Comment by omus** [comment]
Will merge after CI completes

**Comment by omus** [comment]
bors try

**Comment by hannahilea** [comment]
bors try

**Comment by hannahilea** [comment]
bors try

File: test/awss3.jl [hunk 2]
```diff
@@ -97,8 +97,25 @@ function awss3_tests(base_config)
 
     @testset "Object Copy" begin
         config = assume_testset_role("ReadWriteObject"; base_config)
-        s3_copy(bucket_name, "key1"; to_bucket=bucket_name, to_path="key1.copy")
+        result = s3_copy(
+            config, bucket_name, "key1"; to_bucket=bucket_name, to_path="key1.copy"
+        )
+        @test isa(result, LittleDict)
```
Comment (omus) [inline test/awss3.jl:103]:
```suggestion
        @test result isa AbstractDict
```

**Comment by omus** [comment]
bors try

**Review by omus — APPROVED** [review]
We're good here. I made one minor change so I re-triggered CI and will merge after

**Pull request merged**